admin-dev/themes/new-theme/js/components/modal/progress-modal.ts

```diff
@@ -0,0 +1,421 @@
+    this.body.append(errorContainer);
+    // Modal footer element
+    this.footer = document.createElement('div');
+    this.footer.classList.add('modal-footer');
+
+    this.stopProcessingButton = document.createElement('button');
+    this.stopProcessingButton.setAttribute('type', 'button');
+    this.stopProcessingButton.classList.add('btn', 'btn-outline-secondary', 'btn-lg', 'stop-processing');
+    this.stopProcessingButton.innerHTML = 'Stop processing';
+
+    // Appending element to the modal
+    this.footer.append(this.stopProcessingButton, ...params.customButtons);
+    this.content.append(this.footer);
+
+
+    // Modal title element
+    if (params.modalTitle) {
+      this.title = document.createElement('h4');
+      this.title.classList.add('modal-title');
+      this.title.innerHTML = params.modalTitle;
+    }
+
+    // Constructing the modal
+    if (this.title) {
+      this.header.appendChild(this.title);
+    }
+    this.content.append(this.header, this.body);
+  }
+
+  private buildProgressBar(): HTMLElement
+  {
+    let progress = document.createElement('div');
+    progress.setAttribute('style', 'display: block; width: 100%');
+
+    progress.classList.add
+    (
+      'progress',
+      'active',
+    );
+    let progressBar = document.createElement('div');
+    progressBar.classList.add('progress-bar', 'progress-bar-success');
+    progressBar.setAttribute('role', 'progressbar');
+    progressBar.setAttribute('style', 'width: 0%;');
+
+    this.progressDone = document.createElement('div');
+    this.progressDone.setAttribute('style', 'width: 0%');
+
+    this.progressDone.classList.add
+    (
+      'progress-bar',
+      'progress-bar-success',
+    );
+    this.progressDone.setAttribute('role', 'progressbar');
+    this.progressDone.setAttribute('id', 'modal_progressbar_done');
+    progress.append(this.progressDone);
+
+    return progress;
+  }
+}
+export class ProgressModalErrorContainer {
+  footer!: HTMLElement;
+
+  container!: HTMLElement;
+
+  dialog!: HTMLElement;
+
+  content!: HTMLElement;
+
+  message!: HTMLElement;
+
+  header!: HTMLElement;
+
+  title?: HTMLElement;
+
+  body!: HTMLElement;
+
+  switchButton!: HTMLElement;
+
+  downloadErrorsButton!: HTMLElement;
+
+  errorContainer!: HTMLElement;
+
+  /* This constructor is important to force the input type but ESLint is not happy about it*/
+  /* eslint-disable no-useless-constructor */
+  constructor(params: ProgressModalParams) {
+    this.buildModalContainer(params);
+  }
+
+  buildModalContainer(params: ProgressModalParams): void {
+    this.content = document.createElement('div');
+    this.content.classList.add('modal-content');
+
+    // Modal message element
+    this.message = document.createElement('p');
+    this.message.classList.add('modal-message');
+
+    // Modal header element
+    this.header = document.createElement('div');
+    this.header.classList.add('modal-header');
+    this.body = document.createElement('div');
```

Comment:
I believe at least this class is needed for bootstrap modal no?

```suggestion
    this.body = document.createElement('div');
    this.body.classList.add('modal-body');
```